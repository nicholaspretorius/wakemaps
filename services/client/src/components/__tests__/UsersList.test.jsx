import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import UsersList from "./../UsersList";

afterEach(cleanup);

const users = [
  {
    id: 1,
    username: "nicholas",
    email: "nicholaspretorius@gmail.com",
    active: true
  },
  {
    id: 2,
    username: "nic",
    email: "nic@fake.com",
    active: true
  }
];

it("renders", () => {
  const { asFragment } = render(<UsersList users={users} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders a username", () => {
  const { getByText } = render(<UsersList users={users} />);
  expect(getByText("nicholas")).toHaveClass("username");
  expect(getByText("nic")).toHaveClass("username");
});
