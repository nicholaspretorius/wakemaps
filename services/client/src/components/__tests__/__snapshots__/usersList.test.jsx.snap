// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <p
      class="box title is-4 username"
    >
      nicholas
    </p>
    <p
      class="box title is-4 username"
    >
      nic
    </p>
  </div>
</DocumentFragment>
`;
